### PR TITLE
TD-1498 Combined inactive filter and tag issues

### PR DESCRIPTION
**Pull request by martinfoster-hee**
### JIRA link
[TD-1498](https://hee-tis.atlassian.net/jira/software/c/projects/TD/boards/165?modal=detail&selectedIssue=TD-1498)

### Description
Initial load conditions of DelegateCourses and other search/sort/paginate pages were failing under JS due to incorrect setup of tag filters.
Also subsequent issue with combined Archived+Inactive tag filter not working in Typescript implementation.

### Screenshots
Screenshots below.

-----
### Developer checks
I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

[TD-1498]: https://hee-tis.atlassian.net/browse/TD-1498?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

1_delegatecourses_defaultJSload
2_delegatecourses_nofiltersJS
3_delegatecourses_combinedArchived
4_delegatecourses_combinedInactive
5_delegatecourses_active

### CODE REVIEW

File: DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/filter.ts
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/filter.ts:1]: Hi @martinfoster-hee, this filter.ts seems to be a common component used at various places. Are we sure that this is not tampering with other places where this is being used?

**Review by rshrirohit — COMMENTED** [review]
please see inline review for filter.ts

File: DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/filter.ts
Comment (kevwhitt-hee) [reply DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/filter.ts:1]: Is there a way that we could move the bespoke code for the course lists out of the generic `filter.ts` class to keep it generic, @martinfoster-hee?

**Comment by kevwhitt-hee** [comment]
I have tested this and it is working as expected and isn't breaking other implementations of the search, sort, filter, paginate service that don't have Inactive/Archived statuses. 
Happy for this to be merged and moved to test with a note to test both JS and non-JS implementations and other parts of the platform where the service is being used.

**Review by kevwhitt-hee — APPROVED** [review]

**Pull request merged**